### PR TITLE
Add visitors to generated Kotlin code

### PR DESCRIPTION
**Pull request by dlurton**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pig-tests/test/org/partiql/pig/tests/InspectingVisitorTests.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            )
+        }
+        InspectingWalker(visitor).walkTestSumTriplet(node)
+
+        assertEquals(21, visitor.accumulator)
+    }
+
+    // Need tests for...
+    // optional elements
+    // Variadic elements
```
Comment (dlurton) [inline pig-tests/test/org/partiql/pig/tests/InspectingVisitorTests.kt:95]: Also metas

File: pig/src/org/partiql/pig/generator/kotlin/KTypeDomainConverter.kt [hunk 2]
```diff
@@ -298,41 +298,48 @@ internal class KTypeDomainConverter(private val typeDomain: TypeDomain) {
             KProperty(
                 kotlinName = element.identifier.snakeToCamelCase(),
                 tag = element.tag, // TODO: use tag
```
Comment (dlurton) [inline pig/src/org/partiql/pig/generator/kotlin/KTypeDomainConverter.kt:300]: this is done. remove todo

File: pig-tests/test/org/partiql/pig/tests/InspectingVisitorTests.kt (hunk 1 above)
Comment (dlurton) [reply pig-tests/test/org/partiql/pig/tests/InspectingVisitorTests.kt]: done

File: pig-runtime/src/org/partiql/pig/runtime/InspectingDomainVisitorBase.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    open fun visitIonElement(node: IonElement) {
+        // default does nothing
+    }
+
+    open fun visitMetas(node: MetaContainer) {
+        // default does nothing.
+    }
+}
+
+open class InspectingDomainWalkerBase<T: InspectingDomainVisitorBase>(
```
Comment (therapon) [inline pig-runtime/src/org/partiql/pig/runtime/InspectingDomainVisitorBase.kt:40]: To the question in the PR's conversation tab, lets start minimally. Union the two and make the inspecting visitor as simple as possible. If the need arises we can refactor.

**Pull request merged**